### PR TITLE
Added quick navigation

### PR DESCRIPTION
**Pull request by polodasi123**
GoToMenu component with 'Heute' and 'Datum' buttons for fast navigation in CalendarView. Includes a modal date picker for selecting specific weeks and updates CalendarView and its styles to support these features.

<comment on what this PR does>

closes #<issue number>

---

### Implementation

- [x] Does this code change accomplish what it is supposed to do?
- [x] Can this solution be simplified without unreasonable effort
- [x] Is the code modular enough?
- [x] Does similar functionality already exist in the codebase? If yes, why isn’t it reused?
- [x] Single Responsibility Principle: Does any class serve only one purpose? If not, it should probably split up in multiple classes.
- [x] Interface Segregation: If interface implemented, is it fully implemented?
- [x] Dependency Injection: Are central object instantiation processes managed by a single context?
- [x] If any functionality is missing, is it marked with a comment that starts with TODO:?
- [x] Do you see any potential to improve the performance of the code significantly? 

### Logic Errors and Bugs

- [x] Can you think of any use case in which the code does not behave as intended?
- [x] Can you think of any inputs or external events that could break the code?
- [x] Are edge cases covered by tests?

### Error Handling and Logging

- [x] Is error handling done the correct way?
- [x] Should any logging or debugging information be added or removed?
- [x] Are error messages user-friendly?

### Dependencies (internal and external)

- [x] Were updates to documentation, configuration, or readme files made as required by this change?
- [ ] Is the API.md up-to-date?
- [x] Are the enumerations between frontend and backend synchronized?

### Security and Data Privacy

- [ ] Does the code introduce any security vulnerabilities?
- [ ] Is (user) input validated, sanitized, and escaped to prevent security attacks such as cross-site scripting or SQL injection?
- [ ] Is sensitive data like user data handled and stored securely?
- [ ] Is the right encryption used?
- [x] Is data retrieved from external APIs or libraries checked for security issues?
- [x] Were any secrets committed?

### Testing and Testability (later)

- [ ] Have automated tests been added, or have related ones been updated to cover the change?
- [ ] Do the existing tests reasonably cover the code change?
- [ ] Are there some test cases, input or edge cases that should be tested in addition?

### Usability and Accessibility

- [x] Is the proposed solution (UI) accessible (e.g. colorblindness, ...)?
- [x] Is the API/UI intuitive to use?
- [x] Was the app tested on another device?

### Readability

- [x] Is the code easy to understand?
- [x] Which parts were confusing to you and why?
- [x] Can the readability of the code be improved by smaller methods/different names?
- [x] Is the code located in the right file/folder?
- [x] Do you think certain methods should be restructured to have a more intuitive control flow?
- [x] Are there redundant or outdated comments?
- [x] Could some comments convey the message better?
- [x] Could some comments be removed by making the code itself more readable?

### CODE REVIEW

File: VLA-Frontend/src/components/calendar/GoToButton.tsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+      {showDatePicker && (
+        <div className="cv-datePickerOverlay" onClick={handleOverlayClick}>
+          <div className="cv-datePickerBox" onClick={handleBoxClick}>
+            <h3 className="cv-datePickerTitle">Datum auswählen</h3>
+            <input
+              type="date"
+              className="cv-datePickerInput"
+              defaultValue={toISODateLocal(currentWeekStart)}
+              onChange={(e) => handleDateChange(e.target.value)}
```
Comment (CarolinRoell) [inline VLA-Frontend/src/components/calendar/GoToButton.tsx]: Like this, if I try to type in the date using the keyboard, the date is changed after I typed in the day. I would eighter change this so that the date is changed only after I press enter, or that the keyboard input on this is deactivated.

Comment (polodasi123) [reply VLA-Frontend/src/components/calendar/GoToButton.tsx]: Changes on the date are now possible by using the keyboard + added confirmation to jump in time so this mistakes cant happen anymore

**Comment by JohaJung** [comment]
> Mit welchem Browser kannst du die mobile version sehen? 

Firefox hat in den Entwicklertools die Funktion, die Bildschirmgröße anzupassen.

Außerdem kann ich in meinem lokalen Netzwerk problemlos mit meinem Handy auf den von `npm run dev` auf meinem Laptop gestarteten Server zugreifen.

> Und brauchen wir eine mobile version, oder sollen wir uns nur auf desktop fokusieren vielleicht?

Es ist leider von Anfang an Anforderung gewesen, die Software auf allen Geräten benutzen zu können, siehe sein PDF von der Projektausschreibung:
https://bp.st.informatik.tu-darmstadt.de/proposals.html#22-Kremser-VLAOrganisation

**Comment by polodasi123** [comment]
image
Hab gerade herausgefunden, dass das bei Chrome auch geht, falls das jemand braucht. Siehe Screenshot, einfach, wenn man F12 drückt.
